django/db/models/fields/__init__.py

```diff
@@ -1522,6 +1522,9 @@ def get_internal_type(self):
     def to_python(self, value):
         if value is None:
             return value
+        context = decimal.Context(prec=self.max_digits)
```

Comment:
`self.max_digits` should be immutable, so why not create it once in `__init__` and store it as `self._context` ?